pyrsa/rdm/calc.py

```diff
@@ -314,18 +315,15 @@ def calc_rdm_crossnobis(dataset, descriptor, noise=None,
             weights.append(data_test.n_obs)
     else:  # a list of noises was provided
         measurements = []
-        variances = []
         for i_fold in range(len(cv_folds)):
             data = dataset.subset_obs(cv_descriptor, cv_folds[i_fold])
             measurements.append(average_dataset_by(data, descriptor)[0])
-            variances.append(np.linalg.inv(noise[i_fold]))
         for i_fold in range(len(cv_folds)):
-            for j_fold in range(i_fold + 1, len(cv_folds)):
+            for j_fold in range(len(cv_folds)):
                 if i_fold != j_fold:
                     rdm = _calc_rdm_crossnobis_single(
                         measurements[i_fold], measurements[j_fold],
-                        np.linalg.inv(variances[i_fold]
-                                      + variances[j_fold]))
+                        noise[i_fold])
```

Comment:
Made this change because adding 2 variances resulted in reporting each distance as half as large.

Reply:
Hi!
Thanks for placing a pointer to this location.

 Here I fear we need this combination of the two variances.
This is the case where we have different noise variances for the different folds. This combination of the two variances leads to a down weighting whenever one of the two variances is large, i.e. for all combinations where we are unsure about either measurement. After your change we would care only about the noise in one of the two folds (the i_fold).  

Overall scaling should be fine nonetheless. Thus, I would suggest to put the inverse of the sum back in but then multiply the overall result by two.

Reply:
How about if we iterate both `i_fold`,`j_fold` over all fold permutations (changes at line 322)?

I think the math would be equivalent (?) but it would follow the crossnobis equation in `Walther et al. (Neuroimage 2016)` a little more obviously:
![image](https://user-images.githubusercontent.com/3221512/105524721-e1508d80-5c94-11eb-9145-ce5fd70ec90c.png)
(which only uses a single `noise[fold_A]` per split)

Reply:
Unfortunately the math is not equivalent. What you suggest would be:
<html>
<body>
d<sup>2</sup> = (b<sub>j</sub>-b<sub>k</sub>)<sub>A</sub> S<sub>A</sub><sup>-1</sup> (b<sub>j</sub>-b<sub>k</sub>)<sub>B</sub> <sup>T</sup> + (b<sub>j</sub>-b<sub>k</sub>)<sub>A</sub> S<sub>B</sub><sup>-1</sup> (b<sub>j</sub>-b<sub>k</sub>)<sub>B</sub> <sup>T</sup> <br>
= (b<sub>j</sub>-b<sub>k</sub>)<sub>A</sub>( S<sub>A</sub><sup>-1</sup> + S<sub>B</sub><sup>-1</sup> ) (b<sub>j</sub>-b<sub>k</sub>)<sub>B</sub> <sup>T</sup> <br>
&ne;(b<sub>j</sub>-b<sub>k</sub>)<sub>A</sub>( S<sub>A</sub> + S<sub>B</sub> )<sup>-1</sup> (b<sub>j</sub>-b<sub>k</sub>)<sub>B</sub> <sup>T</sup> <br>
</body>
</html>

where the last unequal line would be what the code did before the change. 

Origninally, I implemented a switch to choose how the variances for the two folds should be included, but this made the code here quite messy and I thus decided to reduce it to one variant. This is intentionally slightly different from the Walther et al. formulation. The main reason for this is what I alluded to in my earlier comment: In the Walther et. al formula the summand described will still have a strong influence even if the variance in B was gigantic, which we want to avoid. Thus, I would like to keep the original formulas arguing that they are a better formula than what Walther proposed in this fringe case.

tldr: this is not the same and the difference to Walther et al. is intentional, please keep it.

Reply:
Understood, thanks for the explanation! I'll update this PR.

Reply:
Updated! Looks like Codacy is failing because it isn't tracking the `main` branch? https://app.codacy.com/gh/rsagroup/pyrsa/pullRequest?prid=6800656 @HeikoSchuett